File: tests/.hooks/cli/secondary/internal/sdk/sdkinternal/hooks/customsecurityhook.go
package hooks

import (
	"errors"
	"net/http"

	"openapi/internal/sdk/models/operations"
)

type CustomSecurityHook struct{}

var _ beforeRequestHook = (*CustomSecurityHook)(nil)

func (i *CustomSecurityHook) BeforeRequest(hookCtx BeforeRequestContext, req *http.Request) (*http.Request, error) {
	switch hookCtx.OperationID {
	case "customSchemeAppId":
		if hookCtx.SecuritySource == nil {
			return nil, errors.New("security source is nil")
		}
		sec, err := hookCtx.SecuritySource(req.Context())
		if err != nil {
			return nil, err
		}
		customSchemeAppIDSecurity, ok := sec.(operations.CustomSchemeAppIDSecurity)
		if !ok {
			return nil, errors.New("security source is not of type CustomSchemeAppIDSecurity")
		}
		req.Header.Set("X-Security-App-Id", customSchemeAppIDSecurity.AppID)
		req.Header.Set("X-Security-Secret", customSchemeAppIDSecurity.Secret)
	}

	return req, nil
}

File: tests/.hooks/php/secondary/src/Hooks/CustomSecurityHook.php
<?php

declare(strict_types=1);

namespace alphabetically\early\Hooks;

use alphabetically\early\Models\Operations\CustomSchemeAppIdSecurity;
use Psr\Http\Message\RequestInterface;

class CustomSecurityHook implements BeforeRequestHook
{
    public function beforeRequest(BeforeRequestContext $context, RequestInterface $request): RequestInterface
    {
        $request = $request->withHeader('Idempotency-Key', 'some-key');

        switch ($context->operationID) {
            case 'customSchemeAppId':
                if ($context->securitySource === null) {
                    throw new \InvalidArgumentException('Security source is null');
                }

                $customSecurity = $context->securitySource->call($context);

                if (! $customSecurity instanceof CustomSchemeAppIdSecurity) {
                    throw new \InvalidArgumentException('Security source is not of type CustomSchemeAppIdSecurity');
                }

                $request = $request->withAddedHeader('X-Security-App-Id', $customSecurity->appId);
                $request = $request->withAddedHeader('X-Security-Secret', $customSecurity->secret);
                break;
        }

        return $request;
    }
}
